algorithm/overlay/line_merge.go

```diff
@@ -9,7 +9,7 @@ import (
 //	or an empty atomic geometry, or an empty GEOMETRYCOLLECTION
 func LineMerge(ml matrix.Collection) matrix.Collection {
 	for i := 0; i < len(ml)-1; i++ {
-		for j := 1; j < len(ml); j++ {
+		for j := i + 1; j < len(ml); j++ {
```

Comment:
the length of list `ml` will change.

Reply:
take and done，
todo Rewrite with monotone chain